From 6b869b2b7d7bbaa8efebc6e9d2c4153987027c6a Mon Sep 17 00:00:00 2001
From: Koen Kooi <koen@dominion.thruhere.net>
Date: Wed, 8 Jul 2026 16:49:07 +0200
Subject: [PATCH 25/46] AnnaThermostat: add generic measurement fallback

Map unhandled measurement types to switch/custom/text Domoticz devices based on value semantics.

Signed-off-by: Koen Kooi <koen@dominion.thruhere.net>
Assisted-by: GPT-5.3-Codex

Upstream-Status: Inappropriate [WIP]
---
 hardware/AnnaThermostat.cpp | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/hardware/AnnaThermostat.cpp b/hardware/AnnaThermostat.cpp
index dafb39e62..796c2204b 100644
--- a/hardware/AnnaThermostat.cpp
+++ b/hardware/AnnaThermostat.cpp
@@ -1288,6 +1288,21 @@ void CAnnaThermostat::GetMeterDetails()
 					SendPercentageSensor(applianceIDhex, 0, batterypercentage, valveposition, ApplianceName + " " + sname);
                 }
             }
+			else
+			{
+				if (!tmpstr.empty())
+				{
+					const uint8_t child = static_cast<uint8_t>(40 + (std::hash<std::string>{}(sname) % 90));
+					float num = 0.0F;
+					bool on = false;
+					if (isEnabledState(tmpstr, on))
+						SendSwitch(applianceIDhex, child, batterypercentage, on, 0, ApplianceName + " " + sname, m_Name);
+					else if (isNumericValue(tmpstr, num))
+						SendCustomSensor(applianceIDhex, child, batterypercentage, num, ApplianceName + " " + sname, "");
+					else
+						SendTextSensor(applianceIDhex, child, batterypercentage, tmpstr, ApplianceName + " " + sname);
+				}
+			}
         }
         pAppliance = pAppliance->NextSiblingElement("appliance");
     }
